### ISSUE TITLE
Refinement of ruleset

### ISSUE THREAD

**Issue by gdude2002** [python-discord/meta#21]
Below is a proposition for a new set of rules for our community. The so-called *Python Discord Etiquette* will be written up alongside this later.

Feel free to comment with any feedback.

> We have a small but strict set of rules on our server. Please read over them and take them on board - if you don't understand anything or need some clarification, feel free to ask any staff member! 
> 
> 1. Follow the Discord Community Guidelines
> 2. Follow the Python Discord Etiquette
> 3. We will not help you with anything that might break a law or a terms of service of any kind.
> 4. Listen to and respect the staff members. If they ask you to do something, do it.
> 5. We do not allow advertising of any kind.
> 6. This is an English-speaking server, so please speak English to the best of your ability.
> 
> Please note that these rules are subject to change. This ruleset is not exhaustive, and our moderation staff may choose to take action against situations not related to any of these rules at their discretion.

---

References: #20

**Comment by nottheswimmer** [comment]
I was told to come here and nitpick, so here I am.

- **`we will not help you with anything that might break a law`**
"A law" is too vague. Without specifying particular jurisdictions, this could be improved by specifying that `we will not help you engage in criminal activity` (alt: `illegal activity`) or similar. I think the phrase `criminal activity` provides enough specificity that it bars activities that are illegal both at a server level (e.g. the United States, including both Federal and relevant State laws) and at a local level (e.g. wherever the poster lives), without including irrelevant laws.

- **`or a terms of service of any kind.`**
Same issue in lacking specificity here. Perhaps `or violate a service's terms and conditions where applicable`, although it's a bit wordy.

- **`If they ask you to do something, do it.`**
`Listen to and respect the staff members` may have already been sufficient to get this across.

- **`We do not allow advertising of any kind.`**
I'd argue that you do, and should. The Python Discord is a place for sharing information. In programming, most useful information is a plug (i.e. advertisement) for a relevant product. You should use Celery! You should use Django! [If it has a proper noun, it's pretty much an advertisement](https://www.reddit.com/r/HailCorporate/). I believe the same goes if you're promoting your own projects. "Hey guys, check out this thing I made. *links to git repository*" -- I believe we should be able to do that, but it's definitely an ad. I'd suggest that the rule be changed to a simple `do not spam` -- or if you want to go further, perhaps `do not post with the intent to advertise products which you are affiliated with` or similar.

**Comment by LordHemlock** [comment]
I think that we need to keep Rule 5 as the "Don't do anything against any ToS" thing, as quite a few of our regulars even know Rule 5 by heart.  Changing which spot it'd be in would throw a lot of folks off.

**Comment by scragly** [comment]
> 1. Follow the Discord Community Guidelines  

As the community guidelines already assume the reader has agreed to comply with the official discord terms of service, it never actually says that it should be followed in any of the page's guidelines explicitly. 
As we wish to ensure member's are fully and unequivocally aware that we intend to use TOS compliance as a server rule, we should change this to be:

**1. Follow the [Discord Community Guidelines](https://discordapp.com/guidelines) and [Terms Of Service](https://discordapp.com/terms)**

------
> 2. Follow the Python Discord Etiquette

Some form of generalised idea of what this will contain would be good.

-----
> 3. We will not help you with anything that might break a law or a terms of service of any kind.

The wording of this is kinda clunky, yeah. 

The term law is fine, but it should be "applicable laws" to cover the relevant jurisdictions without actually going into detail. "illegal activity" is the same thing, but awkward wording. "criminal activity" is actually incorrect terminology, as criminal law is only one part of the law, with civil law being disregarded in that case.

The rewording I'm giving also ensures that they realise they themselves are personally not meant to be helping other's with law/tos breaching either, as previously it could be interpreted as only staff cannot personally answer.

I also believe some form of anti-cheat rule should be present here.

**3. Members cannot assist with violating or circumventing any applicable laws or terms of services, nor provide answers for graded coursework/exams.**

----
>4. Listen to and respect the staff members. If they ask you to do something, do it.

The second part seems a bit crass, but I don't mind maybe some rewording:

**4. Listen to and respect staff members, their instructions and any warnings.**

-----
> 5. We do not allow advertising of any kind.

In reference to the previous posts, I'd like to say that recommendation's are not advertisements. A recommendation's most important purpose is to cater to the asker's needs while an advertisement's most important purpose is for the responder to gain something, even if that's just an additional user or maybe some feedback. It's usually relatively easy to see the intention behind an answer, but even if an instance is more ambiguous, mods can use message history and their own judgement to decide what was the most likely case.

I would though try make it less aggressive/more professional, while also including spamming, as that encapsulates things like sending multiple of the same message in a short time, no matter if it's the same channel or across multiple channels.

Also adjusting it to "unapproved advertising" allows staff and members to safely share officially partnered servers and projects, and allows #show-your-project posts to be treated as 'approved advertising'.

**5. Spamming and unapproved advertising is not permitted. Personal projects can be showcased in #show-your-projects.** 

-----
> 6. This is an English-speaking server, so please speak English to the best of your ability.

This is fine.

**Comment by heavysaturn** [comment]
these suggestions are good but some are also very wordy. Some of the intention was to simplify. "circumventing any applicable laws" is a mouthful and probably beyond the language level of some of our younger members, especially those for whom English is a second language.

**Comment by scragly** [comment]
That's fair. What about:
3. Members cannot help with breaking or getting around any laws or terms of services that may apply, and cannot provide answers for graded coursework or exams.

**Comment by heavysaturn** [comment]
yep that's better. not even sure it needs the "that may apply". for the most part I think all your other suggestions are improvements. need to look over them more carefully when I am back at a computer but we should definitely implement some or even most of your suggestions. thanks.

**Comment by syntaxaire** [comment]
After discussion in #mods I want to propose an additional sentence to Rule 5. Current text:
> We will not help you with anything that might break a law or the terms of service of any other community, site, service, or otherwise - No piracy, brute-forcing, captcha circumvention, sneaker bots, or anything else of that nature

Amended text:
> We will not help you with anything that might break a law or the terms of service of any other community, site, service, or otherwise - No piracy, brute-forcing, captcha circumvention, or anything else of that nature. We will not help you scrape e-commerce sites.

Rationale:
- scraping is nearly always forbidden by these sites anyway
- a blanket ban makes moderation easier because we don't have to research Terms and Conditions on tiny retailers

Although the purpose of these bots is invariably to help with sniping and scalping for the grey market, I'm not particularly interested in white knighting for these retailers. My interest is more to make moderation simpler to enforce.
